samples/bluetooth/iso_time_sync/src/controller_time_nrf52.c

```diff
@@ -237,15 +237,15 @@ uint64_t controller_time_us_get(void)
 
 void controller_time_trigger_set(uint64_t timestamp_us)
 {
-	uint32_t num_overflows = timestamp_us / 512000000UL;
-	uint64_t overflow_time_us = num_overflows * num_overflows;
+	uint64_t timestamp_without_rtc_offset =
+		timestamp_us - rtc_ticks_to_us(offset_ticks_and_controller_to_app_rtc);
 
-	uint32_t rtc_remainder_time_us = timestamp_us - overflow_time_us;
-	uint32_t rtc_val =
-		us_to_rtc_ticks(rtc_remainder_time_us) - offset_ticks_and_controller_to_app_rtc;
+	uint32_t num_overflows = timestamp_without_rtc_offset / 512000000UL;
```

Comment:
could this be simplified as 
```suggestion
	uint32_t overflow_time_us = timestamp_without_rtc_offset % 512000000UL;
```